### PR TITLE
Document API stability

### PR DESCRIPTION
**Pull request by carlescufi**
Below is a screenshot of the table that documents stability:

image

After this is merged, I will close https://github.com/zephyrproject-rtos/zephyr/projects/15

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: doc/reference/peripherals/gna.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
+.. _gna_api:
+
+GNA
+###
+
+Overview
+********
+
+The GNA API provides access to Intel's Gaussian Mixutre Model and Neural Network
```
Comment (mbolivar-nordic) [inline doc/reference/peripherals/gna.rst]:
```suggestion
The GNA API provides access to Intel's Gaussian Mixture Model and Neural Network
```

File: doc/reference/audio/dmic.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.. _audio_dmic_api:
+
+DMIC
+####
+
+Overview
+********
+
+The audio dmic interface provides access to digital microphones.
```
Comment (mbolivar-nordic) [inline doc/reference/audio/dmic.rst]:
```suggestion
The audio DMIC interface provides access to digital microphones.
```

**Comment by carlescufi** [comment]
API meeting:

@nashif mentions that an additional column with "Version changed"
@nashif also mentions that we should include updating the table in the release checklist
@pabigot: we should document the deprecation release in the code, alongside the `__deprecated`.

**Review by mbolivar-nordic — APPROVED** [review]
Excellent! I love all the reorganization. I just found a couple of typos.

File: doc/reference/stability.rst [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+   * - :ref:`gna_api`
+     - Experimental
+     - 1.14
+
+   * - :ref:`gpio_api`
+     - Stable
+     - 1.0
+
+   * - :ref:`hwinfo_api`
+     - Unstable
```
Comment (alexanderwachter) [inline doc/reference/stability.rst]: I would like to declare hwinfo as stable.

Comment (carlescufi) [reply doc/reference/stability.rst]: This will be coming soon

**Comment by carlescufi** [comment]
@nashif @mbolivar-nordic added a "Version modified" column

**Comment by mbolivar-nordic** [comment]
> @nashif @mbolivar-nordic added a "Version modified" column

Is there a process for making sure this is updated for each release? If not I still believe this will go stale immediately.

**Comment by carlescufi** [comment]
> > @nashif @mbolivar-nordic added a "Version modified" column
> 
> Is there a process for making sure this is updated for each release? If not I still believe this will go stale immediately.

I've added it to the checklist template.

File: doc/reference/peripherals/gna.rst (hunk 1 above)
Comment (carlescufi) [reply doc/reference/peripherals/gna.rst]: fixed, thanks

File: doc/reference/audio/dmic.rst (hunk 2 above)
Comment (carlescufi) [reply doc/reference/audio/dmic.rst]: fixed, thanks

**Pull request merged**